assembl/static2/package.json

```diff
@@ -54,14 +54,14 @@
     "css-loader": "^1.0.0",
     "empty": "^0.10.1",
     "enzyme": "^3.4.1",
-    "enzyme-adapter-react-16.3": "^1.4.1",
-    "eslint": "^4.6.1",
-    "eslint-config-airbnb": "^15.1.0",
-    "eslint-friendly-formatter": "^3.0.0",
-    "eslint-plugin-flowtype": "^2.35.1",
-    "eslint-plugin-import": "^2.7.0",
-    "eslint-plugin-jsx-a11y": "^5.1.1",
-    "eslint-plugin-react": "^7.3.0",
+    "enzyme-adapter-react-16": "^1.14.0",
```

Comment:
I think, since we are updating libs versions, we can also use strict version for every library in the package.json file instead of `~` or `^` or `~>`. That way we are sure that libraries are 1:1 in every single local machine and remote machine

Reply:
I disagree. You should avoid pinning version in `package.json`. This is what `yarn.lock` is for. You can then use `yarn upgrade-interactive` to see the update and decide to update the packages.

Reply:
Libraries that pin their dependencies is a real problem like `searchkit` that pinned an old version of `lodash` or `react-tags` that pinned everything.

Reply:
FYI, for the `lodash` dependency in `searchkit`, this is overriden with yarn resolutions, see in `package.json` :  ` "resolutions": { "*/lodash": "^4.17.11" }`

Reply:
personally I've never worked in an environment where the deps are NOT pinned. In the exampmle of searchkit and lodash, it might be for a VERY good reason that they have not bumped the version of lodash (probably not though). 

In essence we don't wan't a package to get bumped and it breaks our software. yes the .lock files prevent this but they should not be used as the dependence either. For example, static2 does not build with npm, only yarn because there is no package-lock.json so we have ENFORCED using yarn - that should never happen. 

My rule - fix the deps, do not enforce a build tool (yarn or npm) and do not check in .lock files - you don't need to if you're versions are fixed.

I'm putting in place renovate on the repositories for check for updates in order to stay up to date: https://renovatebot.com/

Reply:
Pinning in `package.json` is not enough because you don't pin versions of dependencies of dependencies. You need a lock file as well.

Reply:
The only way to properly freeze all versions is using `npm ci` (not `npm install`) with `package-lock.json` OR `yarn` with `yarn.lock`. We chose `yarn` for static2 before the implementation of the `package-lock.json` in npm 5 (we were using node 6 with npm 3 at that time, npm 5 wasn't released). Maybe we can switch now to `npm` instead of `yarn` for static2.

Reply:
npm and yarn are a lot closer - I think the dev should be free to choose but we should commit both files and be clear how to build using either case. I think the fabric scripts update the version of npm on your machine and then use yarn for the build anyway. Let's make this outside the scope of this PR unless you want to commit a package-lock.json with it

Reply:
fabric install a specific version of node and npm yes, and it updates it if you don't have those versions. It's using wrongly `npm update` (should really be `npm ci`) for static, and `yarn` for static2. You need to chose between using `npm` or `yarn`, you can't commit both `package-lock.json` and `yarn.lock`, the developer will never do both, the files will be out of sync.

Reply:
Well then, I don't want to block this PR so let's move on and we will adapt later with specific coding rules

Reply:
FYI We currently use the feature yarn workspaces that doesn't exist with npm, and yarn resolutions.